spec/lib/contacts/distributed_lock_spec.rb

```diff
@@ -0,0 +1,26 @@
+require "rails_helper"
+require "redis"
+require "redis-lock"
+
+RSpec.describe "Contacts::DistributedLock" do
+  describe "#lock" do
+    let(:mock) { instance_double("Object") }
+
+    it "delegates to the mock" do
+      expect(mock).to receive(:to_s)
+
+      Contacts::DistributedLock.new.lock do
+        mock.to_s
+      end
+    end
+
+    it "fails silently if the task is already being run and thus held in a Redis lock" do
+      allow(Redis.current).to receive(:lock).and_raise(Redis::Lock::LockNotAcquired)
+      expect(mock).not_to receive(:to_s)
+
+      Contacts::DistributedLock.new.lock do
+        mock.to_s
+      end
+    end
+  end
+end
```

Comment:
This looks good, thanks for putting this in. 

The "it"'s doesn't sound quite right to me. I think they are more like:

- It yields the block when a redis lock can be acquired
- it silently doesn't a yield the block when a lock can't be acquired.

You can use some specific rspec matchers for this: https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/yield-matchers and avoid the need for a mock

e.g:

```
it "yields the block when a redis lock can be acquired" do
  expect(Redis.current).to receive(:lock).and_call_original
  expect { |b| Contacts::DistributedLock.new.lock(&b) }.to yield_control
end
```